### PR TITLE
[New Rule] Sus Network Activity from Unknown Executable

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This rule monitors for network connectivity to the internet from a previously unknown executable located in a suspicious directory to a previously unknown destination ip. An alert from this rule can indicate the presence of potentially malicious activity, such as the execution of unauthorized or suspicious processes attempting to establish connections to unknown or suspicious destinations such as a command and control server. Detecting and investigating such behavior can help identify and mitigate potential security threats, protecting the system and its data from potential compromise.

### Detection
```
host.os.type : "linux" and event.category : "network" and event.action : ("connection_attempted") and 
process.executable : ( 
  /tmp/* or /var/tmp/* or /dev/shm/* or /etc/init.d/* or /etc/rc*.d/* or /etc/crontab or /etc/cron.*/* or 
  /etc/update-motd.d/* or /usr/lib/update-notifier/* or /home/*/.* or /boot/* or /srv/* or /run/* or /etc/rc.local) and
source.ip : (
  127.0.0.0/8 or
  10.0.0.0/8 or
  172.16.0.0/12 or
  192.168.0.0/16 
) and not destination.ip : (
  10.0.0.0/8 or
  127.0.0.0/8 or
  169.254.0.0/16 or
  172.16.0.0/12 or
  192.0.0.0/24 or
  192.0.0.0/29 or
  192.0.0.8/32 or
  192.0.0.9/32 or
  192.0.0.10/32 or
  192.0.0.170/32 or
  192.0.0.171/32 or
  192.0.2.0/24 or
  192.31.196.0/24 or
  192.52.193.0/24 or
  192.168.0.0/16 or
  192.88.99.0/24 or
  224.0.0.0/4 or
  100.64.0.0/10 or
  192.175.48.0/24 or
  198.18.0.0/15 or
  198.51.100.0/24 or
  203.0.113.0/24 or
  240.0.0.0/4 or
  "::1" or
  "FE80::/10" or
  "FF00::/8"
  ) and not process.executable : (
  "/usr/bin/wget" or
  "/usr/bin/curl" or 
  "/usr/bin/apt" or 
  "/usr/bin/dpkg" or 
  "/usr/bin/yum" or
  "/usr/bin/rpm" or
  "/usr/bin/dnf" or
  "/usr/bin/dockerd"
  )

[rule.new_terms]
field = "new_terms_fields"
value = ["destination.ip", "process.executable"]

[[rule.new_terms.history_window_start]]
field = "history_window_start"
value = "now-2d"
```
image

## FP Reduction
By introducing this rule as a new_terms rule, looking at a combination of unknown process executables and destination ips in conjunction with the whitelisting of specific suspicious process.executable locations and the blacklisting of several LOLbins I hope to achieve limited FPs. Rule will be actively monitored for FPs and tuned if necessary. 

## Malware Analysis
These rules are found to be triggered by real malware samples, which @1337-42 and I have been detonating and analysing. Updates will be added to [#105](https://github.com/elastic/elastic-security-labs/issues/105).

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

File: rules/linux/command_and_control_suspicious_network_activity_from_unknown_executable.toml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+directory to a previously unknown destination ip. An alert from this rule can indicate the presence of potentially 
+malicious activity, such as the execution of unauthorized or suspicious processes attempting to establish connections to 
+unknown or suspicious destinations such as a command and control server. Detecting and investigating such behavior can 
+help identify and mitigate potential security threats, protecting the system and its data from potential compromise.
+"""
+from = "now-59m"
+index = ["auditbeat-*", "filebeat-*", "packetbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Suspicious Network Activity to the Internet from Unknown Executable"
```
Comment (w0rk3r) [inline rules/linux/command_and_control_suspicious_network_activity_from_unknown_executable.toml:22]:
```suggestion
name = "Suspicious Network Activity to the Internet by Previously Unknown Executable"
```

**Review by w0rk3r — COMMENTED** [review]
Left some comments

File: rules/linux/command_and_control_suspicious_network_activity_from_unknown_executable.toml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+risk_score = 47
+rule_id = "53617418-17b4-4e9c-8a2c-8deb8086ca4b"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Network", "Command and Control"]
+timestamp_override = "event.ingested"
+type = "new_terms"
+query = '''
+host.os.type : "linux" and event.category : "network" and event.action : ("connection_attempted") and
```
[the remark this answers is not in the document]
Comment (Aegrah) [reply rules/linux/command_and_control_suspicious_network_activity_from_unknown_executable.toml:30]: Fair, changed that in the upcoming commit.

**Review by w0rk3r — APPROVED** [review]
Awesome

**Pull request merged**